enhancements/network/enable-flows-collection.md

```diff
@@ -0,0 +1,307 @@
+                 +-------------+                            
+```
+
+Once the emission has been enabled, the GUI would show a table with the
+status for each node, analogue to the CLI use case:
+
+```text
+                       +-----------+-+
+       Flows emission: |Netflow    |V|
+                       +-----------+-+
+
+                                
+    +------------------------------------------------+
+    |  NAME       STATUS      ENDPOINT               |
+    +------------------------------------------------+
+    |  node-a     emitting    http://10.1.2.3:30001  |
+    |  node-b     emitting    http://10.4.5.6:30001  | 
+    |  node-c     emitting    http://10.7.8.9:30001  |
+    +------------------------------------------------+
+```
+
+### Implementation Details/Notes/Constraints [optional]
+
+In order to ensure an optimal usage of the network, we should restrict flow's
+emission-collection traffic to the local node.
+
+According to the [architectural description of the flows' emitter/collectors pipeline](
+https://docs.google.com/document/d/1kzNfTLXhMolu8VTcH0m9hmvFluN9yKMy7TSNIXJVTrA/edit#heading=h.kulf936ct32p),
+the collector will be deployed as a DaemonSet, to ensure that there will be an
+instance of the collector for each node.
+
+Exposing the collector endpoint as a Kubernetes ClusterIP Service would cause the traffic
+to be distributed across all the nodes (unless you enable the `ServiceInternalTrafficPolicy`
+cluster-wide feature gate, which at this moment is in alpha status, only available
+for the latest Kubernetes 1.21 version).
+
+To limit the flows' traffic to internal nodes, each collector should expose a node
+port, so each OVN instance only has to know the node IP and Port (which might be
+fixed or discovered).
+
+Given the aforementioned details, following subsections depicts the work to do,
+grouped in different tasks.
+
+#### Task 1: modify OVN-Kubernetes configuration to discover automatically the collector address
+
+[Current `Network` operator configuration allows setting a static host:port address](
+https://docs.okd.io/latest/networking/ovn_kubernetes_network_provider/tracking-network-flows.html#nw-network-flows-object_tracking-network-flows).
+This information is forwarded to each OVN instance, on each cluster node. For example:
+
+```yaml
+apiVersion: operator.openshift.io/v1
+kind: Network
+metadata:
+  name: cluster
+spec:
+  exportNetworkFlows:
+    netFlow:
+      collectors:
+        - 192.168.1.99:2056
+```
+
+However, we need to set this address dynamically to ensure that each pod will forward
+the flows to the collector in its same node.
+
+We need to extend the previous configuration to allow replacing the `collectors` property
+by one of the following properties:
+
+* `nodePort: (int)`: if set, the OVN will discover the node IP and submit the flows to the
+  port indicated by such property. E.g. `nodePort: 30023`.
+* `selector: (map)`: if set, the OVN will fetch for any `NodePort` matching
```

Comment:
Can you clarify, this selector is used to match a `Service` object, right? Then to extract its `NodePort`. And further will use this port, but associated with its own host IP, not the IP of the service... is that correct?

Reply:
Yes. It's correct. I'll clarify it in the text